src/main/java/Duke.java

```diff
@@ -1,10 +1,61 @@
+import java.util.List;
+
 public class Duke {
+    private static TaskManager taskManager;
+
+    /**
+     * This is the main method that runs on execution.
+     * 
+     * @param args an array of string inputs
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // intialize variables
+        taskManager = new TaskManager();
+
+        ConversationManager.intializeConversation();
+
+        // try to get user input
+        List<String[]> inputs = ConversationManager.getUserInput();
+
+        // maintain conversation
+        while (!inputs.get(0)[0].equals("bye")) {
+            String command = inputs.get(0)[0];
+            String[] userArgs = inputs.get(1);
+
+            handleCommand(command, userArgs);
+
+            inputs = ConversationManager.getUserInput();
+        }
+
+        ConversationManager.exitConversation();
     }
+
+
+    /**
+     * Handles a command given from user.
+     * 
+     * @param command any string
+     * @param args an array of strings of any length
+     */
+    public static void handleCommand(String command, String[] args) {
+        switch (command) {
+        case "list": {
+            taskManager.listTasks();
+            break;
+        }
+        case "mark": {
+            taskManager.markTask(args);
+            break;
+        }
+        case "unmark": {
+            taskManager.unmarkTask(args);
+            break;
+        }
+        default: {
+            // ConversationManager.printErrorResponse(List.of("command not found"));
+            taskManager.addTask(command, args);
+        }
+        }
```

Comment:
switch statements do not have brackets for each case as per https://se-education.org/guides/conventions/java/basic.html